storage/local/flock/flock.go

```diff
@@ -0,0 +1,37 @@
+// Package flock provides portable file locking. It is essentially ripped out
+// from the code of github.com/syndtr/goleveldb. Strange enough that the
+// standard library does not provide this functionality. Once this package has
+// proven to work as expected, we should probably turn it into a separate
+// general purpose package for humanity.
+package flock
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// Releaser provides the Release method to release a file lock.
+type Releaser interface {
+	Release() error
+}
+
+// New locks the file with the provided name. If the file does not exist, it is
+// created. The returned Releaser is used to release the lock. The returned
+// boolean is true if the file to lock already existed. A non-nil error is
```

Comment:
This could be a valid use-case for named return parameters? `(r Releaser, exists bool, err error)`.